sources/packages/backend/apps/api/src/route-controllers/announcements/_tests_/e2e/announcements.controller.getAnnouncements.e2e-spec.ts

```diff
@@ -0,0 +1,49 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { Repository } from "typeorm";
+import { Announcements } from "@sims/sims-db";
+import { createTestingAppModule } from "../../../../testHelpers";
+
+describe("AnnouncementsController(e2e)-getAnnouncements", () => {
+  let app: INestApplication;
+  let announcementsRepo: Repository<Announcements>;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    announcementsRepo = dataSource.getRepository(Announcements);
+  });
+
+  it("Should return a saved announcement.", async () => {
+    // Arrange
+    const now = new Date();
+    const before = new Date(now.setFullYear(now.getFullYear() - 1));
+    const future = new Date(now.setFullYear(now.getFullYear() + 2));
+
+    const announcement = new Announcements();
```

Comment:
It would be good to have at-least 2 scenarios, announcements returned as current date is within the start date end date range and the opposite. I.e something to be started in future or something which ended already.

Reply:
updated